src/program.cpp

```diff
@@ -805,22 +820,32 @@ cl_build_status cvk_program::compile_source(const cvk_device* device) {
         m_binary_type = CL_PROGRAM_BINARY_TYPE_EXECUTABLE;
     }
 
+    // Translate spirv to llvm ir using llvm-spirv
 #ifdef CLSPV_ONLINE_COMPILER
-    cvk_info("About to compile \"%s\"", build_options.c_str());
-
-    // clspv is based on LLVM. LLVM options parsing is done using global
-    // variable that are not thread safe. Thus, we need to lock call to clspv in
-    // order to ensure a thread safe execution.
-    static std::mutex clspv_compile_lock;
-    clspv_compile_lock.lock();
-    auto result = clspv::CompileFromSourceString(
-        m_source, "", build_options, m_binary.raw_binary(), &m_build_log);
-    clspv_compile_lock.unlock();
+    if (build_from_il) {
+        llvm::LLVMContext Context;
```

Comment:
Looks like you are using llvm naming convention a bit around here, can we use clvk naming convention instead?

Reply:
Thanks I will fix this to be consistent